src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,136 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private StorageDao storageDao;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUP() {
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    void register_loginLessThanMinLength_notOk() {
+        User user = new User();
+        user.setLogin("Bob");
+        user.setPassword("ghjkl543");
+        user.setAge(23);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_userAllInvalidFields_notOk() {
+        User user = new User();
+        user.setLogin("Micael");
+        user.setPassword("ghjkl");
+        user.setAge(15);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_withTheSameFields_notOk() {
+        User user = new User();
+        user.setLogin("Micael");
+        user.setPassword("yxcvb123");
+        user.setAge(18);
+        registrationService.register(user);
+        User user1 = new User();
+        user1.setLogin("Micael");
+        user1.setPassword("yxcvb123");
+        user1.setAge(18);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user1);
+        });
+    }
+
+    @Test
+    void register_withAgeLessThenMin_notOk() {
+        User user = new User();
+        user.setLogin("Micael");
+        user.setPassword("ghjkl1234");
+        user.setAge(12);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
```

Comment:
The test `register_withAgeLessThenMin_notOk` does not cover edge cases for age. It should include tests for exactly the minimum age and one less than the minimum age.